index.js

```diff
@@ -51,9 +86,7 @@ module.exports = function markdownLinkCheck(markdown, opts, callback) {
         if (opts.replacementPatterns) {
             for (let replacementPattern of opts.replacementPatterns) {
                 let pattern = replacementPattern.pattern instanceof RegExp ? replacementPattern.pattern : new RegExp(replacementPattern.pattern);
-                // replace the `{{BASEURL}}` with the opts.projectBaseUrl. Helpful to build absolute urls "relative" to project roots
-                const replacement = replacementPattern.replacement.replace('{{BASEURL}}', opts.projectBaseUrl);
-                link = link.replace(pattern, replacement);
+                link = link.replace(pattern, performSpecialReplacements(replacementPattern.replacement, opts));
```

Comment:
Thanks you for thinking about refactoring this. It could have been done without it but it clearly looks better like that.